esphome/core/application.h

```diff
@@ -72,59 +33,32 @@ class Application {
     this->compilation_time_ = compilation_time;
   }
 
-#ifdef USE_BINARY_SENSOR
-  void register_binary_sensor(binary_sensor::BinarySensor *binary_sensor) {
-    this->binary_sensors_.push_back(binary_sensor);
+  template<class Entity> const std::vector<Entity *> get_entity() {
```

Comment:
Shouldn't this be plural?

```suggestion
  template<class Entity> const std::vector<Entity *> get_entities() {
```